package/gluon-mesh-batman-adv-core/src/respondd.c

```diff
@@ -0,0 +1,606 @@
+                char ifname[IF_NAMESIZE+1];
+                unsigned int flags;
+                struct in6_addr addr;
+                char buf[INET6_ADDRSTRLEN];
+
+                if (sscanf(line,
+                           "%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
+                           "%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
+                           "  %*2x %*2x %*2x %2x %"STRINGIFY(IF_NAMESIZE)"s",
+                           &addr.s6_addr[0], &addr.s6_addr[1], &addr.s6_addr[2], &addr.s6_addr[3],
+                           &addr.s6_addr[4], &addr.s6_addr[5], &addr.s6_addr[6], &addr.s6_addr[7],
+                           &addr.s6_addr[8], &addr.s6_addr[9], &addr.s6_addr[10], &addr.s6_addr[11],
+                           &addr.s6_addr[12], &addr.s6_addr[13], &addr.s6_addr[14], &addr.s6_addr[15],
+                           &flags, ifname) != 18)
+                        continue;
+
+                if (strcmp(ifname, "br-client"))
+                        continue;
+
+                if (flags & (IFA_F_TENTATIVE|IFA_F_DEPRECATED))
+                        continue;
+
+                inet_ntop(AF_INET6, &addr, buf, sizeof(buf));
+
+                json_object_array_add(ret, json_object_new_string(buf));
+        }
+
+        fclose(f);
+        free(line);
+
+        return ret;
+}
+
+static void add_if_not_empty(struct json_object *obj, const char *key, struct json_object *val) {
+        if (json_object_array_length(val))
+                json_object_object_add(obj, key, val);
+        else
+                json_object_put(val);
+}
+
+static bool interface_file_exists(const char *ifname, const char *name) {
+        const char *format = "/sys/class/net/%s/%s";
+        char path[strlen(format) + strlen(ifname) + strlen(name)];
+        snprintf(path, sizeof(path), format, ifname, name);
+
+        struct stat s;
+        return !stat(path, &s);
+}
+
+static void mesh_add_subif(const char *ifname, struct json_object *wireless,
+                           struct json_object *tunnel, struct json_object *other) {
+        struct json_object *address = gluonutil_wrap_and_free_string(gluonutil_get_interface_address(ifname));
+
+        if (interface_file_exists(ifname, "wireless"))
+                json_object_array_add(wireless, address);
+        else if (interface_file_exists(ifname, "tun_flags"))
+                json_object_array_add(tunnel, address);
+        else
+                json_object_array_add(other, address);
+
+}
+
+static struct json_object * get_mesh_subifs(const char *ifname) {
+        struct json_object *wireless = json_object_new_array();
+        struct json_object *tunnel = json_object_new_array();
+        struct json_object *other = json_object_new_array();
+
+        const char *format = "/sys/class/net/%s/lower_*";
+        char pattern[strlen(format) + strlen(ifname)];
```

Comment:
[Again](https://github.com/freifunk-gluon/gluon/pull/612#discussion_r49765063), this should be `- 1`.